ws-client/src/client.rs

```diff
@@ -167,8 +175,14 @@ impl WsClient {
 				max_concurrent_requests,
 			)
 			.await;
+			has_shutdown_sender.store(true, Ordering::Relaxed);
 		});
-		Ok(Self { to_back, request_timeout, error: Mutex::new(ErrorFromBack::Unread(err_rx)) })
+		Ok(Self { to_back, request_timeout, error: Mutex::new(ErrorFromBack::Unread(err_rx)), has_shutdown })
+	}
+
+	/// True if the backend has not shutdown
+	pub async fn is_connected(&self) -> bool {
```

Comment:
missed this one, doesn't need to be async anymore

```suggestion
	pub fn is_connected(&self) -> bool {
```